pkg/operator/ceph/object/spec.go

```diff
@@ -912,3 +922,93 @@ func renderProbe(cfg rgwProbeConfig) (string, error) {
 
 	return writer.String(), nil
 }
+
+func (c *clusterConfig) addDNSNamesToRGWServer() (string, error) {
+	if (c.store.Spec.Hosting == nil) || len(c.store.Spec.Hosting.DNSNames) <= 0 {
+		return "", nil
+	}
+	if !c.clusterInfo.CephVersion.IsAtLeastReef() {
+		return "", errors.New("rgw dns names are supported from ceph v18 onwards")
+	}
+
+	// add default RGW service name to dns names
+	dnsNames := c.store.Spec.Hosting.DNSNames
+	dnsNames = append(dnsNames, domainNameOfService(c.store))
```

Comment:
We don't want to append the service name if there isn't one present.

Reply:
Its for internal rgw server, so service endpoint will be present IMO